### DISCUSSION TITLE
Welcome to hal Discussions!

### DISCUSSION THREAD

**Discussion by diekhans** [ComparativeGenomicsToolkit/hal#181] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by robertopisa** [comment]
Dear Hal developers/maintainers, I have installed and used HAL suite for some analyses on Cactus output in multiple genomes alignment. I am a bit puzzled by the output format of hal- programs: where is it possible to find a detailed explanation of which values are stored in the various columns and how these values are computed? Thank you a lot!

**Comment by glennhickey** [comment]
The documentation I'm aware of is
* [hal paper](https://academic.oup.com/bioinformatics/article/29/10/1341/256598)
* [hal readme](https://github.com/ComparativeGenomicsToolkit/hal/blob/master/README.md)
* [hal section in cactus readme](https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/progressive.md#using-the-hal-output)

If you have a specific question that's not covered above, you can try asking it here.

**Reply by robertopisa** [reply]
Thank you very much for your prompt answer. Unfortunately, I had already checked this documentation: I found in there no explanation about, for instance, what the column "NumSequences" in halStats output files is recording and why it contains an integer larger than 1 only when the referring genome is an ancestor one. The documentation you cited does not contain an explanation of the various output files produced by hal-routines, or, at least, I was unable to find it.

**Comment by glennhickey** [comment]
In HAL each "genome" is a collection of "sequences", a general term for chromosomes or scaffolds or contigs depending on the input.  Ancestral genomes may be broken into several sequences, even if input genomes have 1 each.

**Reply by robertopisa** [reply]
Got it, thank you. But is there a help file where I can find a detailed explanation of each field produced by hal-routines? the readme/help files are often explaining only how to use command line parameters, but they say nothing about the produced output. For instance, what is the meaning of GapDepletion? are they counted with respect to gaps in the current sequence or in the parent one? What I am looking for is a schema that, for each column of the output file of a given routine in the suite, explains what is the variable stored in there and how it is computed, possibly giving the used formula.
